### ISSUE TITLE
Implement Shear transform, see whether it achieves Military (should)

### ISSUE THREAD

**Issue by michaliskambi** [castle-engine/castle-engine#291]
Related to talk with @peardox on Discord.

Implement Shear transform (in a way that doesn't make anything slower, because shear is very very seldom needed):

- https://mathworld.wolfram.com/Shear.html
- https://en.wikipedia.org/wiki/Shear_matrix
- https://philippegroarke.com/blog/2017/02/11/reference-opengl-model-view-projection-matrices/

Check then: Military projection == orthographic projection, with trivial "look downward (0, -1, 0) camera direction", and shear applied? See https://en.wikipedia.org/wiki/Oblique_projection#Military_projection .

**Comment by peardox** [comment]
Some musings on Shear for Oblique...

1) Try having a shear angle which for trivial tesing would simply be ([0-7] / 8) * (2 * pi) allowing projection to be 8th point of compass - these are typically used projections in real-world assets. Of particular note would be JRPG 2D assets.

2) The Shear will probably also work in (-1, 0, 0) and (0, 0, -1) - Cavalier + Cabinet (DOn't know which is which).

3) https://github.com/peardox/CubeExplorer/tree/main/data/Medieval_Village_Pack/Buildings/gltf has a set of models suitable for experimentation. They're CC0 and the author is aware of my uses (and loved my experimental Military sample of Inn.glb)

**Comment by peardox** [comment]
I've added a load of the above to my test repo https://github.com/peardox/CubeExplorer

**Comment by michaliskambi** [comment]
I'm closing this, due to low priority of shear transformation and the big complication it would cause throughout CGE code. 

We have a lot of things dealing with `TCastleTransform` transformation and making assumptions how it works (e.g. `Get/Set[World]View`, physics) . Changing it be something more than translation+rotation+scale would be a significant complication to maintain. Of note is that neither glTF nor X3D decided to support shear too.

I promised to play with shear transformation, but it was low priority and remained a low priority due to very small use-case for it. This issue (making military projection) from @peardox remained the only known real use-case for it (and I understand thar `Orthographic.Stretch`, which we added and is much simpler to maintain, also addressed this, though not in such pretty way). I'm sorry -- the feature request is sensible of course (shear does make sense), but it's just too much work (to do, and later to maintain) for too little gain.
